### PR TITLE
binlog: fix bug in load data

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
## What have you changed?
1. some tiny fix when commit txn for load data
2. remove prewritevalue in commit binlog

## What is the type of the changes? 

Bug fix
1. when data is less than 20000 line, load data will not generate binlog
2. commit binlog shouldn't have prewritevalue
3. the binlog size will be bigger than the last time when load data, and may cause oom, should reset the binlog.

## How has this PR been tested? 

load data less than 20000 line.

## Refer to a related  issue link

https://github.com/pingcap/tidb/issues/7035

### CODE REVIEW

**Comment by shenli** [comment]
@WangXiangUSTC Please fix the CI.

**Comment by WangXiangUSTC** [comment]
@coocood  @jackysp PTAL

File: server/conn.go [hunk 1]
```diff
@@ -822,7 +824,8 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 		}
 		return errors.Trace(err)
 	}
-	return errors.Trace(txn.Commit(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx)))
+
+	return errors.Trace(loadDataInfo.Ctx.RefreshTxnCtx(ctx))
```
Comment (coocood) [inline server/conn.go]: What the difference does this change make?

Comment (jackysp) [reply server/conn.go]: RefreshTxnCtx will start a new transaction. We should not start a new transaction here.

File: server/conn.go [hunk 2]
```diff
@@ -813,6 +814,7 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 	}
 
 	txn := loadDataInfo.Ctx.Txn()
+	txn.SetOption(kv.BypassLatch, true)
```
Comment (jackysp) [inline server/conn.go]: I think it should be set before commit.

File: session/session.go [hunk 3]
```diff
@@ -1322,6 +1322,8 @@ func (s *session) RefreshTxnCtx(ctx context.Context) error {
 		return errors.Trace(err)
 	}
 
+	s.GetSessionVars().TxnCtx.Binlog = nil
```
Comment (jackysp) [inline session/session.go]: Is this root cause of OOM?

Comment (WangXiangUSTC) [reply session/session.go]: I think this is the reason.

Comment (coocood) [reply session/session.go]: We should clear the TxnCtx in `NewTxn`.

Comment (zimulala) [reply session/session.go]: Then, we needn't to change the line of 825 in `server/conn.go`.

**Comment by WangXiangUSTC** [comment]
@jackysp @coocood  PTAL again

File: server/conn.go [hunk 4]
```diff
@@ -822,7 +823,9 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 		}
 		return errors.Trace(err)
 	}
-	return errors.Trace(txn.Commit(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx)))
+
+	txn.SetOption(kv.BypassLatch, true)
+	return errors.Trace(loadDataInfo.Ctx.CommitTxn(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx)))
```
Comment (jackysp) [inline server/conn.go]:
```
return errors.Trace(cc.ctx.CommitTxn(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx)))
```
how about this?

**Comment by coocood** [comment]
LGTM

File: server/conn.go (hunk 4 above)
Comment (coocood) [reply server/conn.go]: Then we don't need to change the `Context` interface.

**Comment by coocood** [comment]
LGTM

**Pull request merged**